### PR TITLE
Updates to developer 'getting started' docs

### PR DESCRIPTION
**Pull request by indirectlylit**
## Summary

A number of updates to the 'getting started' page to help new contributors.

* Reorganized sections to make reading flow from start-to-end more natural for a first-time contributor
* Moved Docker-related instructions to end of file, and moved information about implementation details to _./docker/readme.md_
* Flattened nested header structure a bit to make scanning the left-hand navigation easier
* Clarified instructions related to running servers and linting

| before | after |
|--|--|
| image | image |

## Reviewer guidance

To test docs locally, run `sphinx-autobuild --port 8888 . _build`

Note that due to reorganization of sections, the git diffs are not particularly helpful. It may be more useful to load up the final state and browse through it alongside the current state.

----

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [x] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: docs/getting_started.rst [hunk 1]
```diff
@@ -202,238 +203,181 @@ Alternatively, you can run the devserver with `hot reload <https://vue-loader.vu
...
-        * ``KOLIBRI_PROVISIONDEVICE_SUPERUSERNAME``: default ``devowner``
-        * ``KOLIBRI_PROVISIONDEVICE_SUPERUSERPASSWORD``: default ``admin123``
+To ensure a more efficient workflow, install appropriate editor plugins for Vue.js, ESLint, and stylelint.
 
-    * ``KOLIBRI_HOME``: default ``/kolibrihome``
-    * ``KOLIBRI_HTTP_PORT``: default ``8080``
-    * ``KOLIBRI_LANG``: default ``en``
-    * ``CHANNELS_TO_IMPORT``: comma-separated list of channel IDs (not set by default)
 
+Sample data for development
```
Comment (radinamatic) [inline docs/getting_started.rst]: Suggest changing this to `Import channels and resources for development`.
`Data` may suggest that what is being imported is interaction- and DB-like _data_.

**Review by radinamatic — APPROVED** [review]
Improvements look good to me, just one small suggestion.

File: docs/getting_started.rst (hunk 1 above)
Comment (indirectlylit) [reply docs/getting_started.rst]: The diff makes it confusing to see, but this section now also includes:

> Now you can create users, classes, lessons, etc manually. To auto-generate some sample user data you can also run:
> 
> ```bash
>    kolibri manage generateuserdata
> ```

Comment (radinamatic) [reply docs/getting_started.rst]: Fair enough, could you then change the heading to `Sample resources and data for development` to make it more comprehensive? Thank you!

Comment (indirectlylit) [reply docs/getting_started.rst]: updated!

**Pull request merged**